index.js

```diff
@@ -80,12 +81,20 @@ class Listr {
 		this._checkAll(context);
 
 		let tasks;
-		if (this._options.concurrent === true) {
+		if (this._options.concurrent &&
+				typeof this._options.concurrent !== 'number') {
 			tasks = Promise.all(this._tasks.map(task => runTask(task, context, errors)));
+		} else if (typeof this._options.concurrent === 'number' && this._options.concurrent > 0) {
+			const ctasks = chunks(this._tasks, this._options.concurrent);
+			tasks = ctasks.reduce((promise, tasks) => promise.then(() => {
+				return Promise.all(tasks.map(task => {
+					this._checkAll(context);
+					return runTask(task, context, errors);
+				}));
+			}), Promise.resolve());
```

Comment:
Might be worth using something like [`p-map`](https://github.com/sindresorhus/p-map) to simplify the code here.